src/java.base/share/classes/java/lang/invoke/NativeMethodHandle.java

```diff
@@ -23,34 +23,48 @@
  * questions.
  */
 
+/*
+ * ===========================================================================
+ * (c) Copyright IBM Corp. 2023, 2024 All Rights Reserved
+ * ===========================================================================
+ */
+
 package java.lang.invoke;
 
+import java.lang.foreign.MemorySegment;
+import java.lang.foreign.SegmentAllocator;
+
 import jdk.internal.vm.annotation.ForceInline;
-import jdk.internal.foreign.abi.NativeEntryPoint;
 
 import static java.lang.invoke.LambdaForm.*;
 import static java.lang.invoke.MethodHandleNatives.Constants.LM_TRUSTED;
 import static java.lang.invoke.MethodHandleNatives.Constants.REF_invokeStatic;
 import static java.lang.invoke.MethodHandleStatics.newInternalError;
 
 /**
- * This class models a method handle to a native function. A native method handle is made up of a {@link NativeEntryPoint},
+ * This class models a method handle to a native function. A native method handle is made up of a bounded {@link MethodHandle},
  * which is used to capture the characteristics of the native call (such as calling convention to be used,
  * or whether a native transition is required) and a <em>fallback</em> method handle, which can be used
  * when intrinsification of this method handle is not possible.
  */
 /*non-public*/ final class NativeMethodHandle extends MethodHandle {
-    final NativeEntryPoint nep;
+    /* The native entry point is literally the bounded MethodHandle implemented in OpenJ9. */
+    final MethodHandle nep;
+
+    /* The cache array for the bounded MethodHandle stores MemberName and appendix which are populated
```

Comment:
In these comments, I think "bound" is preferable to "bounded".

Reply:
Fixed everywhere.